### ISSUE TITLE
Interaction.enabledIf(() => boolean)

### ISSUE THREAD

**Issue by thelastnode** [palantir/plottable#2877]
It would be great to be able to conditionally enable an interaction based on a callback, especially for interoperating with frameworks like React.

Example:

    const dragLayer = new Plottable.Components.XDragBoxLayer()
        .onDragEnd((bounds: Plottable.Bounds) => { /* ... */ })
        .enabledIf(() => this.props.dragIsEnabled)

**Comment by bluong** [comment]
Note that this is really useful for configuration of `Interaction`s without needing to worry about order of execution

**Comment by bluong** [comment]
The issues that come along with this feature request is interacting with the `enabled()` endpoint.

One idea that I can think of is allowing `enabled()` to take in a callback.  The problem here is mainly how does the getter return?

**Comment by thelastnode** [comment]
Currently, there are two API points that interact with the enabled state of an interaction: the setter `enable(boolean)` and the getter `enabled`.

I would expect the getter `enabled` to continue to return whether the chart is current enabled or not - regardless of whether it's a call to the function passed to `enabledIf` or the value passed to `enable(boolean)`.

For the setter, I think using `enable(boolean)` should replace the behavior of `enabledIf`, and vice versa. I can't think of a real use case for them to be anything but mutually exclusive.

**Comment by bluong** [comment]
However, for the getter evaluating the function, it then gets quickly annoying if a user wanted to get out the object that they used to set.

**Comment by thelastnode** [comment]
You're right, this API doesn't expose a way to get the callback function back again. Is there a case when a user would want that?
